test/src/testPubSub.cpp

```diff
@@ -262,5 +274,7 @@ TEST_F(TestPubSub, interthread) {
 
 int main(int argc, char **argv) {
     ::testing::InitGoogleTest(&argc, argv);
-    return RUN_ALL_TESTS();
+    auto results = RUN_ALL_TESTS();
+    common_instance = nullptr;
+    return results;
```

Comment:
Revert this change. Set up and tear down tasks should be done in the `SetUp()`, `TearDown()`, `SetUpTestSuite()`, and `TearDownTestSuite()` methods in the `TestPubSub` class.